tests/test_metrics.py

```diff
@@ -0,0 +1,55 @@
+from typing import Self
+
+import pytest
+import pytest_asyncio
+from draive import ctx
+from draive.scope.metrics import CombinableScopeMetric, ScopeMetrics, TokenUsage
+
+
+class ExpMetric(CombinableScopeMetric):
+    def __init__(self, value: float = 1) -> None:
+        self._value = value
+
+    def combine_metric(
+        self,
+        other: Self,
+        /,
+    ) -> Self:
+        return self.__class__(value=self._value * other._value)
+
+    def metric_summary(self) -> str | None:
+        return (
+            f"exp value: {self._value}"
+        )
+
+@pytest_asyncio.fixture
+async def scope_metrics() -> ScopeMetrics:
+    async with ctx.new():
+        await ctx.record(ExpMetric(value=1))
+
+        async with ctx.nested("child"):
+            await ctx.record(TokenUsage(input_tokens=44, output_tokens=55))
+
+            async with ctx.nested("grandchild_1"):
+                await ctx.record(TokenUsage(input_tokens=444, output_tokens=555))
+                await ctx.record(ExpMetric(value=5))
+
+            async with ctx.nested("grandchild_2"):
+                await ctx.record(TokenUsage(input_tokens=222, output_tokens=333))
+                await ctx.record(ExpMetric(value=7))
+
+        return ctx.current_metrics()
+
+@pytest.mark.asyncio
+async def test_combinable_metrics(scope_metrics: ScopeMetrics) -> None:
```

Comment:
there is still no test for the case where child has metric which parent dos not have, please add this one